src/simulators.jl

```diff
@@ -438,3 +438,121 @@ function B_step!(s, dt_eff, acceleration_vector, compute_forces::Bool, n_threads
+
+arguments:
+- `dt::S`: the time step of the simulation.
+- `temperatures::TP`: the temperatures corresponding to the replicas.
+- `simulators::ST`: individual simulators for simulating each replica.
+- `exchange_time::ET`: the time interval between replica exchange attempt.
+"""
+struct TemperatureREMD{N, T, S, DT, TP, ST, ET}
+    dt::DT
+    temperatures::TP
+    simulators::ST
+    exchange_time::ET
+end
+
+function TemperatureREMD(;
+                dt,
+                temperatures,
+                simulators,
+                exchange_time,
+                kwargs...)
+    S = eltype(simulators)
+    T = eltype(temperatures)
+    N = length(temperatures)
+    DT = typeof(dt)
+    TP = typeof(temperatures)
+    ET = typeof(exchange_time)
+    if length(simulators) != length(temperatures)
+        throw(ArgumentError("Number of temperatures must match number of simulators"))
+    end
+    if exchange_time <= dt
+        throw(ArgumentError("Exchange time must be greater than the time step"))
+    end
+    simulators = Tuple(simulators[i] for i in 1:N)
+    ST = typeof(simulators)
+    
+    return TemperatureREMD{N, T, S, DT, TP, ST, ET}(dt, temperatures, simulators, exchange_time)
+end
+
+function simulate!(sys::ReplicaSystem,
+                    sim::TemperatureREMD,
+                    n_steps::Int;
+                    assign_velocities::Bool=false,
+                    rng=Random.GLOBAL_RNG,
+                    n_threads::Int=Threads.nthreads())
+    if sys.n_replicas != length(sim.simulators)
+        throw(ArgumentError("Number of replicas in ReplicaSystem and simulators in TemperatureREMD do not match."))
+    end
+
+    if n_threads > sys.n_replicas
+        thread_div = equal_parts(n_threads, sys.n_replicas)
+    else # pass 1 thread per replica
+        thread_div = equal_parts(sys.n_replicas, sys.n_replicas)
+    end
+
+    # calculate n_cycles and n_steps_per_cycle from dt and exchange_time
+    n_cycles = convert(Int64, (n_steps * sim.dt) ÷ sim.exchange_time)
+    cycle_length = n_steps ÷ n_cycles
+    remaining_steps = n_steps % n_cycles
+
+    if assign_velocities
+        for i in eachindex(sys.replicas)
+            random_velocities!(sys.replicas[i], sim.temperatures[i])
+        end
+    end
+
+    for cycle=1:n_cycles
+        @sync for idx in eachindex(sim.simulators)
+            Threads.@spawn Molly.simulate!(sys.replicas[idx], sim.simulators[idx], cycle_length; n_threads=thread_div[idx])
+        end
+
+        if cycle != n_cycles
+            n = rand(1:sys.n_replicas)
+            m = mod(n, sys.n_replicas) + 1
+            k_b = sys.k
+            T_n, T_m = sim.temperatures[n], sim.temperatures[m]
+            β_n, β_m = 1/(k_b*T_n), 1/(k_b*T_m)
+            V_n, V_m = potential_energy(sys.replicas[n]), potential_energy(sys.replicas[m])
+            Δ = ustrip((β_m - β_n)*(V_n - V_m))
```

Comment:
`ustrip` not required?

Reply:
Here `Δ` is giving me units of `mol^-1`. So, the comparison below doesn't work. I think I should log the value of `Δ` with these units as well.

Reply:
That `mol^-1` should be divided out, it's a little fiddly but the following should work. Replace `k_b = sys.k` with
```julia
if dimension(sys.energy_units) == u"𝐋^2 * 𝐌 * 𝐍^-1 * 𝐓^-2"
    k_b = sys.k * T(Unitful.Na)
else
    k_b = sys.k
end
```
and `sys::ReplicaSystem` in the function arguments with `sys::ReplicaSystem{D, G, T}` ... ` where {D, G, T}`.

This will probably change the acceptance rate in the examples.

Reply:
Okay I have done this, the number of exchanges also changed (increased). In the above example I am getting about 30 exchanges (from 40 attempts) maybe because the temperature difference is small.